youtube_dl/extractor/kaltura.py

```diff
@@ -117,7 +117,9 @@ def _extract_url(webpage):
                         (?P<q1>['"])wid(?P=q1)\s*:\s*
                         (?P<q2>['"])_?(?P<partner_id>(?:(?!(?P=q2)).)+)(?P=q2),.*?
                         (?P<q3>['"])entry_?[Ii]d(?P=q3)\s*:\s*
-                        (?P<q4>['"])(?P<id>(?:(?!(?P=q4)).)+)(?P=q4)(?:,|\s*\})
+                        (?P<q4>['"])(?P<id>(?:(?!(?P=q4)).)+)(?P=q4)(?:,.*?
+                        (?P<q5>['"])ks(?P=q5)\s*:\s*
+                        (?P<q6>['"])(?P<ks>(?:(?!(?P=q6)).)+)(?P=q6))?(?:,|\s*\})
```

Comment:
This breaks all non ks embeds. ks part must be optional.

Reply:
Er, it shouldn't break non-ks embeds, unless I'm rustier at Python regular expressions than I thought? Note that the ks part is wrapped in (?: )?, which should make it optional. I tested it against both types and didn't run into any issues, it would just set q5, q6, and ks to None in the match object for embeds without ks.

Reply:
This is unreadable. Indent optional group properly.

Reply:
Done. The original regular expression and its neighbors aren't particularly "readable" anyway, so what would be considered "proper" was unclear, but I hope you'll be happy with the latest commit.

Reply:
The original regex is clear line by line expression without any optional parts.